GVFS/GVFS.Common/GVFSEnlistment.cs

```diff
@@ -238,5 +251,26 @@ private void CreateHiddenDirectory(string path)
             DirectoryInfo dir = Directory.CreateDirectory(path);
             dir.Attributes = FileAttributes.Hidden;
         }
+
+        private string GetId(string key)
+        {
+            GitProcess.Result configResult = this.GetGitProcess().GetFromLocalConfig(key);
+            if (!configResult.HasErrors)
+            {
+                return configResult.Output.Trim();
+            }
+
+            return string.Empty;
+        }
+
+        private GitProcess GetGitProcess()

```

Comment:
This opens an interesting question about how we are using the `GitProcess` in other places since in most of those places we pass the `Enlistment` it might make sense to just grab this one that is in the enlistment object.  But that would be for a future PR.